<?xml version="1.0"?>
<launch>

  <param name="/topic_prefix" value="/oakd"/> 
  <param name="/fps_rgb_yolo" value="30.0"/> <!-- data rate -->
  <param name="/fps_stereo_depth" value="30.0"/> <!-- data rate -->
  <param name="/fps_IMU" value="500"/> <!-- data rate, integer -->

  <!-- image type - getting raw and compressed both is possible -->
  <param name="/get_raw" value="true"/> 
  <param name="/get_compressed" value="false"/> 

  <!-- use or not -->
  <param name="/get_imu" value="true"/> 
  <param name="/get_rgb" value="false"/> 

  <param name="/get_stereo_ir" value="true"/> 
  <param name="/get_stereo_depth" value="false"/> 
  <param name="/get_pointcloud" value="false"/>

  <!-- depth post processing -->
  <param name="/pcl_max_range" value="6.0"/> <!-- [m] further than this value will not be published -->
  <param name="/depth_confidence" value="175"/>  <!-- 0~255 -->
  <param name="/bilateral_sigma" value="3"/>  <!-- 0 ~ 65535, depth filter -->
  <param name="/use_spatialFilter" value="false"/> 
      <param name="/spatialFilter_holefilling_radius" value="4"/> 
      <param name="/spatialFilter_iteration_num" value="1"/> 
      <param name="/spatialFilter_alpha" value="0.5"/> 
  <param name="/use_temporalFilter" value="false"/> 
      <param name="/temporalFilter_alpha" value="0.4"/> 
  <param name="/use_speckleFilter" value="false"/> 
      <param name="/speckleFilter_range" value="50"/> 

<!-- setCoordinateSize, setAnchors, setAnchorMasks should be directly edited in main.h file -->
  <param name="/get_YOLO" value="false"/>
  <param name="/thread_num" value="1"/>  <!-- best is 2 -->
  <param name="/blob_file" value="/blob_files/tiny-yolo-v4.blob"/>
  <param name="/class_file" value="/blob_files/class.txt"/>
  <param name="/infer_img_width" value="416"/>
  <param name="/infer_img_height" value="416"/> 
  <param name="/class_num" value="80"/> 
  <param name="/confidence_threshold" value="0.4"/> 
  <param name="/iou_threshold" value="0.3"/> 

  <node name="oakd_ros_node" pkg="oakd_ros" type="oakd_ros_node" output="screen"/>
  <!-- launch-prefix="gdb -ex run --> <!--args" -->

</launch>
